### ISSUE TITLE
StopIteration raised on truncated data

### ISSUE THREAD

**Issue by schmidtc** [tebeka/fastavro#46]
The following will raise "StopIteration" instead of a "more" valid Exception...

```
import fastavro
from StringIO import StringIO
fastavro.load(StringIO(), {'fields':[{'type':'boolean', 'name':'a'}], 'type':'record', 'name': 'test'})
```

This can cause data loss when using fastavro in a for loop. For example when coupled with python's kafka client which process messages in batches, a single bad message in the batch will cause all following messages to be discarded with no exceptions or warnings thrown. The message need not be empty, truncated messages can also trigger the error if the truncation occurs at the boundary of two values.

**Comment by tebeka** [comment]
Thanks for reporting!

I need to think about it. StopIteration *is* the way to signal end of data in Python iterators - see [here](https://docs.python.org/2/library/exceptions.html#exceptions.StopIteration).

However looks like in your case the load should return an empty iterator or raise an error about not being able to read the header. I'll work on that.

BTW: I'm getting a different error when running your code: `error: unpack requires a string argument of length 1`. Which version of fastavro and Python are you using?

**Comment by schmidtc** [comment]
Using fastavro version 0.9.5 with python 2.7

We lost 16% of the messages in one of our of production systems without realizing it because of this issue. Granted it was a threading bug on our end that was corrupting the messages in the first place.  Processing messages in a loop is a very common use case, aborting that loop with no warning or exception results in a very difficult to debug system.

Agreed that StopIteration is the correct signal to stop an iterator, however `fastavro.load` is not appear to be an iterator. Regardless malformed data should raise a valid exception.

**Comment by tebeka** [comment]
Fixed in 0.9.9 (should be up soon). Reading from empty stream will raise EOFError
